subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -2023,15 +2014,88 @@ static int lwm2m_delete_handler(struct lwm2m_engine_obj *obj,
 				     context->path->obj_inst_id);
 }
 
+static int get_coap_packet_len(struct net_pkt *pkt)
+{
+	struct net_buf *frag;
+	u16_t offset;
+	u16_t len;
+
+	frag = net_frag_read_be16(pkt->frags,
+				  net_pkt_ip_hdr_len(pkt) +
+				  net_pkt_ipv6_ext_len(pkt) +
+				  4, &offset, &len);
+	if (!frag && offset == 0xffff) {
+		return -1;
+	}
+
+	return len - NET_UDPH_LEN;
+}
+
+#define COAP_MARKER 0xFF
```

Comment:
Shall we keep one define in include/net/coap.h for this purpose? because subsys/net/lib/coap/coap.c has also one internal define.

Reply:
We could add this to include/net/coap.h.   However, this define and the following 2 functions go away in a future patch once I've converted the backend LwM2M content formatters to use multi fragments instead of a contiguous buffer.